### PR TITLE
Feat: s3 del bucket exists logic

### PR DESCRIPTION
**Pull request by swartz-k**
**What this PR does**:
Delete S3 Bucket Exists logic to support more storage like AlibabaClous OSS

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: modules/ingester/instance.go [hunk 1]
```diff
@@ -332,7 +332,17 @@ func (i *instance) tracesToCut(cutoff time.Duration, immediate bool) []*trace {
 func (i *instance) writeTraceToHeadBlock(id common.ID, b []byte) error {
 	i.blocksMtx.Lock()
 	defer i.blocksMtx.Unlock()
-
+	if i.headBlock == nil {
```
Comment (joe-elliott) [inline modules/ingester/instance.go:335]: Any particular reason you added this?  I would prefer keeping the PR focused on the change to remove BucketExists.

File: tempodb/wal/append_block.go [hunk 2]
```diff
@@ -89,6 +90,9 @@ func (h *AppendBlock) Complete(w *WAL, combiner common.ObjectCombiner) (*encodin
 }
 
 func (h *AppendBlock) Find(id common.ID, combiner common.ObjectCombiner) ([]byte, error) {
+	if h.appender == nil {
```
Comment (joe-elliott) [inline tempodb/wal/append_block.go:93]: Same comment here.

**Review by joe-elliott — CHANGES_REQUESTED** [review]

**Comment by annanay25** [comment]
Agree with Joe on this. Let's focus on removing the `BucketExists` call in this PR, and we can review the other changes in a different PR.

File: modules/ingester/instance.go (hunk 1 above)
Comment (swartz-k) [reply modules/ingester/instance.go:335]: Yes. I will meet some CI errors if those code not add.

Comment (joe-elliott) [reply modules/ingester/instance.go]: Ah, I think @mdisibio discovered and is working on this issue.  Thank you for also trying to address it.

**Comment by swartz-k** [comment]
> Agree with Joe on this. Let's focus on removing the `BucketExists` call in this PR, and we can review the other changes in a different PR.

Right. I misunderstand it. Sometimes CI will fail.

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**